src/main/scala/fi/oph/koski/raportit/PerusopetuksenOppijamäärätRaportti.scala

```diff
@@ -0,0 +1,142 @@
+      virheellisestiSiirretytMuutKuinVaikeimminVammaiset = r.<<,
+      erityiselläTuella = r.<<,
+      majoitusetu = r.<<,
+      kuljetusetu = r.<<,
+      sisäoppilaitosmainenMajoitus = r.<<,
+      koulukoti = r.<<,
+      joustavaPerusopetus = r.<<
+    )
+  )
+
+  def build(oppilaitosOids: List[String], date: Date)(implicit u: KoskiSession): DataSheet = {
+    val raporttiQuery = query(validateOrganisaatioOids(oppilaitosOids), date).as[PerusopetuksenOppijamäärätRaporttiRow]
+    val rows = runDbSync(raporttiQuery, timeout = 5.minutes)
+    DataSheet(
+      title = "Suoritukset",
+      rows = rows,
+      columnSettings = columnSettings
+    )
+  }
+
+  private def query(oppilaitosOids: List[String], date: Date)(implicit u: KoskiSession) = {
+    sql"""
+    with q as (
+      select
+        r_opiskeluoikeus.oppilaitos_nimi,
+        r_koodisto_koodi.nimi as opetuskieli,
+        r_paatason_suoritus.koulutusmoduuli_koodiarvo as vuosiluokka,
+        count(*) as oppilaita,
+        count(case when aidinkieli != 'fi' and aidinkieli != 'sv' and aidinkieli != 'se' and aidinkieli != 'ri' and aidinkieli != 'vk' then 1 end) as vieraskielisiä,
+        count(case when pidennetty_oppivelvollisuus and vaikeasti_vammainen then 1 end) as pidennettyOppivelvollisuusJaVaikeastiVammainen,
+        count(case when pidennetty_oppivelvollisuus and not vaikeasti_vammainen and vammainen then 1 end) as pidennettyOppivelvollisuusJaMuuKuinVaikeimminVammainen,
+        count(case when not pidennetty_oppivelvollisuus and vaikeasti_vammainen and erityinen_tuki then 1 end) as virheellisestiSiirretytVaikeastiVammaiset,
+        count(case when not pidennetty_oppivelvollisuus and not vaikeasti_vammainen and vammainen and erityinen_tuki then 1 end) as virheellisestiSiirretytMuutKuinVaikeimminVammaiset,
+        count(case when erityinen_tuki then 1 end) as erityiselläTuella,
+        count(case when majoitusetu then 1 end) as majoitusetu,
+        count(case when kuljetusetu then 1 end) as kuljetusetu,
+        count(case when sisaoppilaitosmainen_majoitus then 1 end) as sisäoppilaitosmainenMajoitus,
+        count(case when koulukoti then 1 end) as koulukoti,
+        count(case when joustava_perusopetus then 1 end) as joustava_perusopetus
+      from r_opiskeluoikeus
+      join r_henkilo on r_henkilo.oppija_oid = r_opiskeluoikeus.oppija_oid
+      join r_opiskeluoikeus_aikajakso aikajakso on aikajakso.opiskeluoikeus_oid = r_opiskeluoikeus.opiskeluoikeus_oid
+      join r_organisaatio_kieli on r_organisaatio_kieli.organisaatio_oid = oppilaitos_oid
+      join r_koodisto_koodi on r_koodisto_koodi.koodisto_uri = split_part(split_part(kielikoodi, '#', 1), '_', 1) and r_koodisto_koodi.koodiarvo = split_part(kielikoodi, '#', 2)
+      join r_organisaatio on r_organisaatio.organisaatio_oid = oppilaitos_oid
+      left join r_paatason_suoritus on r_paatason_suoritus.opiskeluoikeus_oid = r_opiskeluoikeus.opiskeluoikeus_oid
```

Comment:
päätason suorituksia voi olla useita, oonks väärässä vai eiks täst tuu karteesinen tulo mikä vaikuttaa sit noihin laskuihin väärällä tavalla?

aikajaksoja pitäisi olla vaan "yks rivi per päivä" joten sitä ongelmaa ei pitäisi tulla.

Reply:
eiks tän näkis jos lisäis testin?

Reply:
eiku hetkinen eiks toi group_by hoida tän?

Reply:
Hmm siis useita päätason suorituksia per oppija ja vuosiluokka?

Reply:
jos oppilas jää luokalle niin siinä tapauksessa vois ehkä olla enemmän kun yks vuosiluokan suoritus, ehkä tätä pitää kans kysyy?